### PR TITLE
fix: fetch required containerd versions from components.json during e2e tests

### PR DESCRIPTION
**Pull request by UtheMan**
Now that components are automatically updated by Renovate, we will try to automate how we get the expected versions of packages in e2e tests, to avoid extra manual work.

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

**Review by Devinwong — APPROVED** [review]
lgtm. Approved. Will be great to run a complete test vhd build and e2e tests to validate.

File: e2e/scenario_test.go [hunk 1]
```diff
@@ -519,7 +519,7 @@ func Test_ubuntu1804(t *testing.T) {
 			Cluster: ClusterKubenet,
 			VHD:     config.VHDUbuntu1804Gen2Containerd,
 			LiveVMValidators: []*LiveVMValidator{
-				containerdVersionValidator("1.7.1+azure-1"),
+				containerdVersionValidator(getExpectedPackageVersions("containerd", "ubuntu", "r1804")[0]),
 				runcVersionValidator("1.1.14-1"),
```
Comment (djsly) [inline e2e/scenario_test.go:523]: Could you do the same for runc ?

**Review by djsly — CHANGES_REQUESTED** [review]

**Review by djsly — APPROVED** [review]

File: e2e/scenario_test.go (hunk 1 above)
Comment (UtheMan) [reply e2e/scenario_test.go]: Added, but there are differences between what `--version` command returns for containerd/runc and what we store in `components.json`. Will sync with Devin on Monday to discuss

File: e2e/scenario_test.go [hunk 2]
```diff
@@ -23,7 +23,9 @@ func Test_azurelinuxv2(t *testing.T) {
 				nbc.AgentPoolProfile.Distro = "aks-azurelinux-v2-gen2"
 			},
 			LiveVMValidators: []*LiveVMValidator{
-				containerdVersionValidator(getExpectedPackageVersions("containerd", "azurelinux", "current")[0]),
+				// Get the ExpectedPackageVersions from components.json->containerd->mariner instead of azurelinux intentionally.
+				// Aadhar Agarwal confirmed that AzureLinuxV2 reporting itself as `Mariner` is expected behavior because of a transition plan.
+				containerdVersionValidator(getExpectedPackageVersions("containerd", "mariner", "current")[0]),
```
Comment (Devinwong) [inline e2e/scenario_test.go:28]: @UtheMan I took your branch and made this change. Please see comments.

File: parts/linux/cloud-init/artifacts/components.json [hunk 3]
```diff
@@ -658,7 +658,7 @@
           }
         },
         "azurelinux": {
-          "current": {
+          "v3.0": {
```
Comment (Devinwong) [inline parts/linux/cloud-init/artifacts/components.json:661]: Bear with me, I should have created another branch to modify this. Now I modified the schema to support `v3.0` for explicit Azure Linux v3.

File: parts/linux/cloud-init/artifacts/README-COMPONENTS.md [hunk 4]
```diff
@@ -107,6 +107,7 @@ Please refer to [components.cue](../../../../schemas/components.cue) for the mos
 	current?: #ReleaseDownloadURI
 }
 #AzureLinuxOSDistro: {
+	"v3.0"?:  #ReleaseDownloadURI
```
Comment (Devinwong) [inline parts/linux/cloud-init/artifacts/README-COMPONENTS.md:110]: modified the schema to support v3.0 for explicit Azure Linux v3.

Comment (UtheMan) [reply parts/linux/cloud-init/artifacts/README-COMPONENTS.md]: Oh could you add those changes in a separate PR? I didn't notice they were there and i force pushed to this branch for clean up, I apologize

Comment (Devinwong) [reply parts/linux/cloud-init/artifacts/README-COMPONENTS.md]: no worries. I should have let you know first. It's not a lot of changes. I will create a separate PR.

File: parts/linux/cloud-init/artifacts/components.json [hunk 5]
```diff
@@ -590,7 +590,7 @@
             "versionsV2": [
               {
                 "renovateTag": "<DO_NOT_UPDATE>",
-                "latestVersion": "1.1.9-5.cm2"
+                "latestVersion": "1.1.9-7.cm2"
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/components.json:593]: we're gonna remove runc entries for mariner/azlinux right?

Comment (Devinwong) [reply parts/linux/cloud-init/artifacts/components.json:593]: I will remove it in my PR

**Review by cameronmeissner — APPROVED** [review]

File: e2e/validators.go [hunk 6]
```diff
@@ -248,35 +248,28 @@ func UlimitValidator(ulimits map[string]string) *LiveVMValidator {
...
-			}
-			return nil
-		},
+// can be used to validate version of moby components like moby-containerd or moby-runc
+func mobyComponentVersionValidator(component, version, packageManager string) *LiveVMValidator {
+	installedCommand := "list --installed"
+	if packageManager == "dnf" {
+	    installedCommand = "list installed"
+	}
+	if packageManager == "apt" {
```
Comment (djsly) [inline e2e/validators.go:257]: redundant logic ?

File: e2e/scenario_test.go [hunk 7]
```diff
@@ -937,7 +939,7 @@ func Test_ubuntu2204ContainerdURL(t *testing.T) {
 				nbc.ContainerdPackageURL = "https://packages.microsoft.com/ubuntu/22.04/prod/pool/main/m/moby-containerd/moby-containerd_1.6.9+azure-ubuntu22.04u1_amd64.deb"
 			},
 			LiveVMValidators: []*LiveVMValidator{
-				containerdVersionValidator("1.6.9"),
+				mobyComponentVersionValidator("containerd", "1.6.9", "apt"),
```
Comment (djsly) [inline e2e/scenario_test.go:942]: not sure I like this hidden contract `"apt"` shouldnt we pass the OS Family instead ?  from config.VHDUbuntu2204Gen2Containerd.OSSKU ? or something ?

**Comment by UtheMan** [comment]
Bypassing, failures in vhd build unrelated to e2e code changes

**Pull request merged**